src/sentry/static/sentry/app/utils/discover/genericDiscoverQuery.tsx

```diff
@@ -152,6 +152,10 @@ class GenericDiscoverQuery<T, P> extends React.Component<Props<T, P>, State<T>>
 
     this.setState({isLoading: true, tableFetchID});
 
+    if (setError) {
+      setError(undefined);
+    }
```

Comment:
Not sure what the preference is but I've seen both

```javascript
setError?.();
```

and

```javascript
callIfFunction(setError);
```

be used in this situation.

Reply:
We were using `callIfFunction` a bunch before typescript added `?.`. However, `?.` is better now as it is shorter and doesn't require imports.

Reply:
right right, I keep forgeting `?.` is a thing